### PR TITLE
mscgen: fix build

### PR DESCRIPTION
**Pull request by siriobalmelli**
###### Motivation for this change

Found `nix-env -iA "nixpkgs.mscgen"` build was broken because of missing `pkg-config` dependency. Added it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by calbrecht — CHANGES_REQUESTED** [review]
Why are there changes in pkgs howdoi? Seems like this should go into a different PR.

**Comment by siriobalmelli** [comment]
> Why are there changes in pkgs howdoi? Seems like this should go into a different PR.

My apologies. They were in fact part of https://github.com/NixOS/nixpkgs/pull/85579 and I screwed up my branching.

Removed these entirely.

**Review by calbrecht — CHANGES_REQUESTED** [review]
`bison`, `pkg-config` and `flex` should go into `nativeBuildInputs` instead.

**Comment by calbrecht** [comment]
Then there is this notice when the tests beeing run
```
Warning: Optional UTF-8 byte-order-mark detected at start of input, but mscgen
         was not configured to use FreeType for text rendering.  Rendering of
         UTF-8 characters in PNG output may be incorrect.
```
Since you added `freetype` you might want to add 
```
configureFlags = [ "--with-freetype" ]
```
and fix the  `Error: gdoTextWidth: Problem doing text layout (GDFONTPATH=)`
or simply leave `freetype` out of dependencies.

**Comment by calbrecht** [comment]
And the title of this PR should read like "mscgen: <your description>".

File: pkgs/tools/graphics/mscgen/default.nix [hunk 1]
```diff
@@ -12,7 +24,7 @@ stdenv.mkDerivation {
     sha256 = "3c3481ae0599e1c2d30b7ed54ab45249127533ab2f20e768a0ae58d8551ddc23";
   };
 
-  buildInputs = [ flex bison gd libjpeg libpng freetype zlib libwebp ];
+  buildInputs = [ bison flex freetype gd libjpeg libpng libwebp pkg-config zlib ];
```
Comment (Mic92) [inline pkgs/tools/graphics/mscgen/default.nix:27]: bison, flex and pkg-config should be in `nativeBuildInputs`.

Comment (calbrecht) [reply pkgs/tools/graphics/mscgen/default.nix:27]: o.O :)

**Comment by calbrecht** [comment]
And last, i guess. Consider adding `outputs = [ "out" "man" ];`. This way the man pages are linked correctly.

**Comment by siriobalmelli** [comment]
> And last, i guess. Consider adding `outputs = [ "out" "man" ];`. This way the man pages are linked correctly.

Thank you; added in 3a7dca99

**Comment by siriobalmelli** [comment]
> `bison`, `pkg-config` and `flex` should go into `nativeBuildInputs` instead.

Thank you, fixed in 3a7dca99

**Comment by siriobalmelli** [comment]
> Then there is this notice when the tests beeing run
> 
> ```
> Warning: Optional UTF-8 byte-order-mark detected at start of input, but mscgen
>          was not configured to use FreeType for text rendering.  Rendering of
>          UTF-8 characters in PNG output may be incorrect.
> ```
> 
> Since you added `freetype` you might want to add
> 
> ```
> configureFlags = [ "--with-freetype" ]
> ```
> 
> and fix the `Error: gdoTextWidth: Problem doing text layout (GDFONTPATH=)`
> or simply leave `freetype` out of dependencies.

Thank you.
I gave this a shot but no quick win in sight, so removed freetype dependency entirely in 3a7dca99

**Review by calbrecht — APPROVED** [review]
Good, thank you :+1:

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 85580` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
<br>- asciidoc-full-with-plugins
<br>- mscgen
</details>

**Review by Mic92 — APPROVED** [review]

**Pull request merged**